### PR TITLE
Added routes for getting courses

### PR DESCRIPTION
**Pull request by Jatewo**
## Description

Added middleware to handling user authentication upon accessing user-specific content. Changed/added routes for courses along with an overhaul on the course model. 

## Changes

Changes made:
  - Created file for getting fake courses (for testing)
  - Removed leftover console.logs
  - Reworked requireLogin to check that the right user is accessing user specific content
  - Added adminOnly middleware for routes only meant to be reachable for admins (note that this is a placeholder and should not be considered as a finished feature)
  - Added tests for middleware
  - Added route for getting all courses (adminOnly)
  - Added route for getting all courses made by a specific content creator
  - Added tests for routes
  - Updated some old tests for the reworked requireLogin
  - Updated the course model

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Notes for Reviewers

Commented out some old tests for a 204: No content result when trying to access a non-existent user. Right now a 401: Unauthorized will returned. We need to discuss how we want to handle this.

Need a review of the code that use course model, as we have changed the model quite a bit.

### CODE REVIEW

**Review by Mafusn — APPROVED** [review]
Looks great

File: models/Courses.js
[the remark this answers is not in the document]
Comment (JackSMunn) [reply models/Courses.js:1]: No, as of right now we do not have a "content creator model".. So the creator field refers to the user model. 

I have deleted one of the sections

[the remark this answers is not in the document]
Comment (JackSMunn) [reply models/Courses.js:1]: Might have to recheck the user/creator problem when we have merged. As of now, all tests pass.

**Review by sofiiagran — APPROVED** [review]
thumbs up, looks good, we should discuss with the onboarding team on how they use the content creator model

**Pull request merged**